src/Tools/BuildValidator/BuildConstructor.cs

```diff
@@ -264,15 +282,20 @@ private static VisualBasicCompilationOptions CreateVisualBasicCompilationOptions
             VB.LanguageVersion langVersion = default;
             VB.LanguageVersionFacts.TryParse(langVersionString, ref langVersion);
 
-            var preprocessorSymbols = string.IsNullOrEmpty(define)
-                ? Array.Empty<KeyValuePair<string, object>>()
-                : define.Split(',')
-                    .Select(s => s.Split('='))
-                    .Select(a => new KeyValuePair<string, object>(a[0], a[1]))
-                    .ToArray();
+            IReadOnlyDictionary<string, object>? preprocessorSymbols = null;
+            if (OptionToString(CompilationOptionNames.Define) is string defineString)
+            {
+                preprocessorSymbols = VisualBasicCommandLineParser.ParseConditionalCompilationSymbols(defineString, out var diagnostics);
```

Comment:
should we just allow this diagnostic bag to propagate out, rather than passing one in?

Reply:
I started with than and changed the return to `(Compilation? Compilation, DiagnosticBag? DiagnosticBag)` but I felt the consumption code was very awkward to write. Eventually I went with the approach of passing a bag down as we do in other parts of the code and it felt a bit more natural.